src/xb-builder-node.c

```diff
@@ -297,6 +301,41 @@ xb_builder_node_parse_literal_text (XbBuilderNode *self, const gchar *text, gssi
 	return g_string_free (tmp, FALSE);
 }
 
+/**
+ * xb_builder_node_tokenize_text:
+ * @self: a #XbBuilderNode
+ *
+ * Tokenize text added with xb_builder_node_set_text().
+ *
+ * Since: 0.3.1
+ **/
+void
+xb_builder_node_tokenize_text (XbBuilderNode *self)
+{
+	XbBuilderNodePrivate *priv = GET_PRIVATE (self);
+	g_auto(GStrv) tokens = NULL;
+	g_auto(GStrv) ascii_tokens = NULL;
+
+	g_return_if_fail (XB_IS_BUILDER_NODE (self));
+
+	if (priv->text == NULL)
+		return;
+	tokens = g_str_tokenize_and_fold (priv->text, NULL, &ascii_tokens);
+	for (guint i = 0; tokens[i] != NULL; i++) {
```

Comment:
```suggestion
	for (gsize i = 0; tokens[i] != NULL; i++) {
```

(Array indices should be `gsize` as they could address all locations in memory if the array is big enough)